src/use-swr.ts

```diff
@@ -64,19 +64,39 @@ if (!IS_SERVER && window.addEventListener) {
     }
   }
 
-  // focus revalidate
-  window.addEventListener(
-    'visibilitychange',
-    () => revalidate(FOCUS_REVALIDATORS),
-    false
-  )
-  window.addEventListener('focus', () => revalidate(FOCUS_REVALIDATORS), false)
-  // reconnect revalidate
-  window.addEventListener(
-    'online',
-    () => revalidate(RECONNECT_REVALIDATORS),
-    false
-  )
+  const hasWindow = typeof window !== 'undefined'
+  const hasDocument = typeof document !== 'undefined'
+  if (hasWindow) {
```

Comment:
I think the 2 conditions doesn't have to be nested?

```
if (hasDocument) {
   // focus revalidators
}
if (hasWindow) {
   // other revalidators
}
```

Reply:
RN environment also have `document` object like the web? 
I think that if RN doesn't have `document` object we should use `window.addEventListener`

Reply:
nope. it has window object but not document. 
I double check again locally with a RN app. it shows that window doesn't have `addEventListener`. and we need to check addEventListener. looks like your first commit also works well and it's shorter

I'll approve it on my side if can be reverted to the previous commit. sorry for my mistake and thanks for the question.

Reply:
Aha. Is this logic ONLY for web, not RN?

Reply:
yes, swr supports browser runtime by default. since react is been ported to different rumtime, people are trying adopt swr along with react apps. I was wondering if it's possible to make swr more flexible to be adopted among runtimes. seems not easy.